google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemTest.java

```diff
@@ -159,4 +162,22 @@ public void testListFiles() throws IOException {
       assertThat(got).containsExactlyElementsIn(goodPaths);
     }
   }
+
+  @Test
+  public void testMatcher() throws IOException {
+    try (FileSystem fs = CloudStorageFileSystem.forBucket("bucket")) {
+      List<String> paths = ImmutableList.of("a.java", "a.text", "folder/c.java", "d");
+      String pattern1 = "glob:*.java";
+      PathMatcher matcher1 = fs.getPathMatcher(pattern1);
+      List<Boolean> matches1 = ImmutableList.of(true, false, true, false);
+      String pattern2 = "glob:*.{java,text}";
+      PathMatcher matcher2 = fs.getPathMatcher(pattern2);
+      List<Boolean> matches2 = ImmutableList.of(true, true, true, false);
+      for (int i=0; i<paths.size(); i++) {
+        Path input = fs.getPath(paths.get(i)).getFileName();
+        assertWithMessage(pattern1 + " on " + paths.get(i)).that(matcher1.matches(input)).isEqualTo(matches1.get(i));
+        assertWithMessage(pattern2 + " on " + paths.get(i)).that(matcher2.matches(input)).isEqualTo(matches2.get(i));
```

Comment:
This approach of comparing items by iterating over a list seems somewhat roundabout. I would prefer a more straightforward approach like this, even if it is more repetitive:

PathMatcher javaFileMatcher = fs.getPathMatcher(pattern1);
...
assertThat(javaFileMatcher.matches("a.java")).isTrue();
assertThat(javaFileMatcher.matches("d")).isFalse();
...